python/cugraph-pyg/cugraph_pyg/examples/taobao_mnmg.py

```diff
@@ -501,23 +506,38 @@ def create_loader(data_l):
             batch["user", "item"].edge_label_index,
         )
         break
-    print(f"Initialized model on rank {global_rank}")
+    init_end = perf_counter()
+    print(
+        f"Initialized model on rank {global_rank}, "
+        f"took {init_end - init_start:.4f} seconds."
+    )
 
     model = DistributedDataParallel(model, device_ids=[local_rank])
     optimizer = torch.optim.Adam(model.parameters(), lr=args.lr)
 
+    training_start = perf_counter()
     best_val_auc = 0
     for epoch in range(1, args.epochs + 1):
+        train_start = perf_counter()
         print("Train")
         loss = train(model, optimizer, train_loader)
+        train_end = perf_counter()
 
         if global_rank == 0:
             print("Val")
+        val_start = perf_counter()
```

Comment:
I think it's safer to synchronize before perf_counter. `train()` should already sync implicitly due to the backward call, but I am not sure about `test()`.